commit 293e86998748bae2d7a8179ad198ec3452f0d472
Author: Richard Braun <rbraun@sceen.net>
Date:   Wed Jun 1 23:19:19 2016 +0200

    Fix pipe_send() with no data
    
    * libpipe/pipe.c (pipe_send): Set value pointed to by amount to 0 if
    if nothing is written.

diff --git a/libpipe/pipe.c b/libpipe/pipe.c
index c3d2a28..0b53921 100644
--- a/libpipe/pipe.c
+++ b/libpipe/pipe.c
@@ -319,7 +319,10 @@ pipe_send (struct pipe *pipe, int noblock, void *source,
 
   /* Nothing to do.  */
   if (data_len == 0 && control_len == 0 && num_ports == 0)
-    return 0;
+    {
+      *amount = 0;
+      return 0;
+    }
 
   err = pipe_wait_writable (pipe, noblock);
   if (err)
